hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -6,7 +6,17 @@ import is.hail.expr.types.virtual._
 import is.hail.table.Ascending
 import is.hail.utils._
 
+import scala.collection.mutable
+
+
 object PruneDeadFields {
+
+  case class ComputeMutableState(requestedType: Memo[BaseType], relationalRefs: mutable.HashMap[String, ArrayBuilder[Type]]) {
+    def rebuildState: RebuildMutableState = RebuildMutableState(requestedType, mutable.HashMap.empty)
+  }
+
+  case class RebuildMutableState(requestedType: Memo[BaseType], relationalRefs: mutable.HashMap[String, Type])
```

Comment:
These classes have strange names.  They are verbs but classes are normally nouns.  Why is compute have an array of types for compute?

Reply:
Ah, is this the mutable state of the compute and rebuild steps, respectively?  Compute here means the memo step?

Reply:
yes, exactly.

Reply:
are you asking about the ArrayBuilder[Type]? These are the usages of the relational refs.